python-backend/app/api/now_applications/models/activity_detail/activity_detail_base.py

```diff
@@ -25,21 +25,22 @@ class ActivityDetailBase(AuditMixin, Base):
     width = db.Column(db.Integer)
     length = db.Column(db.Integer)
     depth = db.Column(db.Integer)
+    height = db.Column(db.Integer)
     quantity = db.Column(db.Integer)
     incline = db.Column(db.Numeric(14, 2))
     incline_unit_type_code = db.Column(db.String, db.ForeignKey('unit_type.unit_type_code'))
     cut_line_length = db.Column(db.Integer)
+    # TODO: cut_line_unit_type_code?
     water_quantity = db.Column(db.Integer)
     water_quantity_unit_type_code = db.Column(db.String, db.ForeignKey('unit_type.unit_type_code'))
 
-    activity_summaries = db.relationship(
-        'ActivitySummaryBase', secondary='activity_summary_detail_xref')
+    _etl_activity_details = db.relationship('ETLActivityDetail', load_on_pending=True)
+
     activity_type_code = db.column_property(
-        db.select(
-            [ActivitySummaryBase.activity_type_code],
-            and_(
-                ActivitySummaryDetailXref.activity_summary_id == ActivitySummaryBase.
-                activity_summary_id,
-                ActivitySummaryDetailXref.activity_detail_id == activity_detail_id)).as_scalar())
+        db.select([ActivitySummaryBase.activity_type_code],
+                  and_(
+                      ActivitySummaryDetailXref.activity_summary_id == ActivitySummaryBase.
+                      activity_summary_id, ActivitySummaryDetailXref.activity_detail_id ==
+                      activity_detail_id)).limit(1).as_scalar())
```

Comment:
Are you able to speak to why you need to `limit(1)` this? Is it just to change the shape of the data?

Reply:
We're querying the type code from the related summary here, but it's through an xref so the detail is potentially related to multiple summary records so we only need one of them. 

Per design, a detail can only be related to summaries of a single type. This is perfectly enforced in code by the polymorphic types. If we wanted to add additional safety around this, we could add this as a constraint at the data level as well.